entity-framework/core/querying/raw-sql.md

```diff
@@ -28,13 +28,16 @@ Raw SQL queries can be used to execute a stored procedure.
 <!-- [!code-csharp[Main](samples/core/Querying/Querying/RawSQL/Sample.cs)] -->
 ``` csharp
 var blogs = context.Blogs
-    .FromSql("EXECUTE dbo.GetMostPopularBlogs")
+    .FromSqlRaw("EXECUTE dbo.GetMostPopularBlogs")
     .ToList();
 ```
 
 ## Passing parameters
 
-As with any API that accepts SQL, it is important to parameterize any user input to protect against a SQL injection attack. You can include parameter placeholders in the SQL query string and then supply parameter values as additional arguments. Any parameter values you supply will automatically be converted to a `DbParameter`.
+As with any API that accepts SQL, it is important to parameterize any user input to protect against a SQL injection attack. `FromSqlRaw` performs simple string interpolation, so any parameters will be included as-is and must be manually sanitized; not doing so may expose your application to major security risks. To safely pass parameters, `FromSqlInterpolated` allows you to include placeholders in the SQL query string and then supply parameter values as additional arguments. Any parameter values you supply will automatically be converted to a `DbParameter` for safe execution. Using parameters with `FromSqlRaw` can still useful for dynamically building your SQL query, but should be used with care and never when simple parameterization is needed.
+
+> [!WARNING]
+> Prior to version 3.0, `FromSqlRaw` and `FromSqlInterpolated` were two overloads named `FromSql`. These two overloads have been obsoleted for safety reasons.
```

Comment:
More than including a warning here, can we describe the original `FromSql` under a "using previous versions" subtitle and have a warning explaining briefly in which cases it was ambiguous/hard to predict what things would get parameterized?

Reply:
Will do.